### PR TITLE
Update tekton operator deps

### LINKED ISSUE #178: [FEATURE] Update to Kustomize v5
### Is there an existing feature request for this?

- [X] I have searched the existing feature requests

### Is your feature request related to a problem or use-case? Please describe.

Kustomize v5 includes several new features, plus breaking API changes that lead up to `kustomize v1` specs. Its first release was way back in February 2023.

### Describe the solution that you would like.

Upgrade the build pipeline to use Kustomize v5. Most notable is the new syntax for `patches` - see https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv5.0.0 for list of breaking changes.

Potentially related to #139 .

### Describe alternatives you have considered.

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by adambkaplan**
# Changes

Update Tekton and k8s dependencies to most recent community supported versions, with related updates to code toolchains:

1. Golang to v1.22
2. k8s to 1.30.z
3. Tekton Operator to v0.74.z

Prerequisite for #230 

Fixes #178 

# Submitter Checklist

- [x] Includes tests if functionality changed/was added
- [x] Includes docs if changes are user-facing
- [x] [Set a kind label on this PR](https://prow.k8s.io/command-help#kind)
- [x] Release notes block has been filled in, or marked NONE

See [the contributor guide](https://github.com/shipwright-io/build/blob/main/CONTRIBUTING.md)
for details on coding conventions, github and prow interactions, and the code review process.

# Release Notes

```release-note
action required: Minimum validated Tekton Operator version is v0.74.0; minimum validated Kubernetes version is v1.30.6. Other tooling such as Kustomize and controller-gen also updated to k8s 1.30 compatible versions.
```

### CODE REVIEW

**Review by SaschaSchwarze0 — CHANGES_REQUESTED** [review]

File: go.mod [hunk 1]
```diff
@@ -10,13 +12,13 @@ require (
 	github.com/onsi/ginkgo/v2 v2.19.0
 	github.com/onsi/gomega v1.33.1
 	github.com/shipwright-io/build v0.13.0
-	github.com/tektoncd/operator v0.67.0
-	k8s.io/api v0.27.11
-	k8s.io/apiextensions-apiserver v0.27.11
-	k8s.io/apimachinery v0.27.11
+	github.com/tektoncd/operator v0.74.0
+	k8s.io/api v0.30.1
```
[the remark this answers is not in the document]
Comment (adambkaplan) [reply go.mod:16]: they just cut 1.30.8 last week :smile_cat: - will update.

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,8 @@
 module github.com/shipwright-io/operator
 
-go 1.22.0
+go 1.22.5
+
+toolchain go1.23.3
```
[the remark this answers is not in the document]
Comment (adambkaplan) [reply go.mod]: done!

**Comment by adambkaplan** [comment]
@SaschaSchwarze0 updated per your feedback - ptal

**Review by SaschaSchwarze0 — APPROVED** [review]
/approve
/lgtm

**Pull request merged**